outbound/adapter/kafka-outbound-publisher.ts

```diff
@@ -0,0 +1,29 @@
+import * as TE from "fp-ts/TaskEither";
+import { flow, pipe } from "fp-ts/lib/function";
+import * as RA from "fp-ts/ReadonlyArray";
+import * as O from "fp-ts/Option";
+import { OutboundPublisher } from "../port/outbound-publisher";
+import { KafkaProducerCompact } from "../../utils/kafka/KafkaProducerCompact";
+import * as KP from "../../utils/kafka/KafkaProducerCompact";
+
+export const create = <T>(
+  producer: KafkaProducerCompact<T>
+): OutboundPublisher<T> => ({
+  publish: (document: T): TE.TaskEither<Error, T> =>
+    pipe(
+      [document],
```

Comment:
So from now on we will publish one document at the time? 
It does not affect performances?

Reply:
IMHO the single commit is a specific implementation of chunks so we can consider a single document like a chunk of one element 😄

Reply:
I agree so I'd rather define publish in a way like this: `(documents: readonly T[]): TE.TaskEither<Error,  readonly T[]>`

Reply:
We could add a new `publishAll` implementation on `OutboundPublisher`

Reply:
I've changed the implementation to keep the "chunks" approach.